pipeline/ingestor/ingestor_common/src/main/scala/uk/ac/wellcome/platform/ingestor/common/IndexConfigOps.scala

```diff
@@ -0,0 +1,22 @@
+package uk.ac.wellcome.platform.ingestor.common
+
+import com.typesafe.config.Config
+import uk.ac.wellcome.elasticsearch.{IndexConfig, RefreshInterval}
+
+trait IndexConfigOps {
+  implicit class IndexConfigOps[IC <: IndexConfig](indexConfig: IC) {
+    def withRefreshIntervalFromConfig(config: Config): IndexConfig = {
+      val isReindexing = config.getBoolean(s"es.is_reindexing")
+      val interval =
+        if (isReindexing) RefreshInterval.Off else indexConfig.refreshInterval
+
+      object NewConfig extends IndexConfig {
+        override def mapping = indexConfig.mapping
+        override def analysis = indexConfig.analysis
+        override def shards = indexConfig.shards
+        override def refreshInterval = interval
+      }
+      NewConfig
+    }
```

Comment:
This feels like something that should be handled by making `IndexConfig` a case class.

Reply:
I wasn't sure how universally useful it was going to be - happy to move it to scala-libs if you think it would be?

Reply:
So my concern is not so much the specific method, more the blatting of distinct types – all we know is that we get an `IndexConfig` out of this method.

If we can do that here and it's fine, it feels like maybe we should be using case classes instead of trying to shoehorn a `copy`-like method onto a trait.

Reply:
Yea, thought the same. My understanding is we're using objects as that's all they are, we never really use typechecking etc.

I'll give it a go to move it over to case classes - as I think that's the more relevant structure as it's the same type with different values.

Reply:
Here's a first pass at the case class implementation:
https://github.com/wellcomecollection/scala-libs/pull/117